src/common/tracer.h

```diff
@@ -0,0 +1,80 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include<cstring>
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+#include<atomic>
+
+#include <jaegertracing/Tracer.h>
+#include <jaegertracing/net/IPAddress.h>
+#include <jaegertracing/net/Socket.h>
+
+
+typedef std::unique_ptr<opentracing::Span> Span;
+
+struct Jaeger_Tracer{
+    Jaeger_Tracer(){}
+    ~Jaeger_Tracer();
+  std::shared_ptr<opentracing::v3::Tracer> tracer = NULL;
+};
+
+//will be used throughout ceph to create spans
+extern Jaeger_Tracer tracer;
+extern std::atomic<bool> jaeger_initialized;
+
+
+static inline void init_tracer(const char* tracerName,const char* filePath){
+      if(jaeger_initialized) return;
+
+      try{
+          auto yaml = YAML::LoadFile(filePath);
+          auto configuration = jaegertracing::Config::parse(yaml);
+
+          jaegertracing::net::Socket socket;
+          socket.open(AF_INET, SOCK_STREAM);
+          const std::string serverURL = configuration.sampler().samplingServerURL();
+          socket.connect(serverURL); // this is used to check if the tracer is able to connect with server successfully
+
+          tracer.tracer = jaegertracing::Tracer::make(
+          tracerName,
+          configuration,
+          jaegertracing::logging::consoleLogger());
+      }catch(...) {return;}
```

Comment:
please follow the our brackets/block [coding style](https://github.com/ceph/ceph/blob/master/CodingStyle)

Reply:
noted